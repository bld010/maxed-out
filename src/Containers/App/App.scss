.App {


  & header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 10px;
    min-height: 80px;
    height: 80px;

    & h1 {
      font-size: 50px;
      margin: 0 50px;
      text-transform: lowercase;
      color: rgb(98, 84, 129);
    }
  }

  & .nav {
    color: black;
    font-size: 20px;
    padding: 20px;
  }

  & main {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    border-top: 5px solid rgb(98, 84, 129);
    border-bottom: 5px solid rgb(98, 84, 129);
    background-color: rgba(98, 84, 129, 0.3);
    padding: 50px;

    & .search-and-disambiguation {
      margin-right: 40px;
      border-radius: 8px; 
      padding: 25px;
      min-width: 25vw;
      background: white;
      min-height: 200px;
      overflow: auto;
      max-height: 700px;
      font-size: 15px;
      filter: drop-shadow(1px 1px 5px #333);

      & h2 {
        text-align: center;
        color: rgb(98, 84, 129)
      }
    }

    & .display {
      background: white;
      border-radius: 8px;
      padding: 20px;
      min-width: 68vw;
      min-height: 700px;
      max-height: 700px;
      width: auto;
      filter: drop-shadow(1px 1px 5px #333);

      & h3 {
        color: rgb(98, 84, 129);
      }
    }
  }

  & .howTo {
    margin-left: 20px;
    display: flex; 
    flex-direction: row;

    & .faq {
      min-width: 40%;
      border: 1px solid black;
    }
  }

  

}
