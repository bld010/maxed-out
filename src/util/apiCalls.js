export const searchCandidateByName = async (searchTerm) => {

  let formattedSearchTerm = searchTerm.toLowerCase().split().join('%20')
  let url = `https://api.open.fec.gov/v1/candidates/search/?sort_null_only=false&sort_nulls_last=false&per_page=100&sort=name&api_key=x6zpz92JgOnDxuca5Vf6QGJIV46FkTVYMvAfBNGl&sort_hide_null=false&name=${formattedSearchTerm}&page=1`


  try {
    let candidateSearchResult = await fetch(url)
    if (candidateSearchResult.ok === false) {
      throw new Error('There was an error searching for the candidate')
    }
    let results = await candidateSearchResult.json()
    return(results.results)
  } catch (err) {
    throw err
  }
}

export const searchCommitteeById = async (committee_id) => {
  let url = `https://api.open.fec.gov/v1/committee/${committee_id}/?sort=name&sort_null_only=false&api_key=x6zpz92JgOnDxuca5Vf6QGJIV46FkTVYMvAfBNGl&per_page=100&sort_nulls_last=false&sort_hide_null=true&page=1`
  try {
    let committeeSearchResult = await fetch(url);
    if (committeeSearchResult.ok === false) {
      throw new Error('There was an error finding the requested committee')
    }
    let results = await committeeSearchResult.json();
    return(results.results)
  } catch(err) {
    throw err
  }

}

export const searchCandidateById = async (candidate_id) => {

  let url = `https://api.open.fec.gov/v1/candidates/search/?api_key=x6zpz92JgOnDxuca5Vf6QGJIV46FkTVYMvAfBNGl&per_page=50&candidate_id=${candidate_id}&sort_nulls_last=false&sort_hide_null=false&page=1&sort=name&sort_null_only=false`

  try {
    let candidateSearchResult = await fetch(url);
    if (candidateSearchResult.ok === false) {
      throw new Error ('There was an error searching for the candidate')
    }
    let results = await candidateSearchResult.json();
    return results.results
  } catch (err) {
    throw err
  }
}

export const fetchPACContributions = async (committee_id) => {

  let url = `https://api.open.fec.gov/v1/schedules/schedule_a/?sort_null_only=false&committee_id=${committee_id}&min_amount=5000&per_page=50&api_key=x6zpz92JgOnDxuca5Vf6QGJIV46FkTVYMvAfBNGl&is_individual=false&sort_hide_null=false&sort=contribution_receipt_date&two_year_transaction_period=2020`

  try {
    let pacContributions = await fetch(url);
    if (pacContributions.ok === false) {
      throw new Error ('There was an error getting the PAC contributions')
    }
    let results = await pacContributions.json();
    return results.results
  } catch (err) {
    throw err
  }
}

export const fetchIndividualContributions = async (committee_id) => {

  let url = `https://api.open.fec.gov/v1/schedules/schedule_a/?sort_null_only=false&committee_id=${committee_id}&min_amount=2800&per_page=50&api_key=x6zpz92JgOnDxuca5Vf6QGJIV46FkTVYMvAfBNGl&is_individual=true&sort_hide_null=false&sort=contribution_receipt_date&two_year_transaction_period=2020`

  try {
    let individualContributions = await fetch(url);
    if (individualContributions.ok === false) {
      throw new Error ('There was an error getting the PAC contributions')
    }
    let results = await individualContributions.json();
    return results.results
  } catch (err) {
    throw err
  }
}