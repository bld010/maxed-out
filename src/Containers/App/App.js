import React, { Component } from 'react';
import './App.scss';
import SearchForm from '../SearchForm/SearchForm';
import { connect } from 'react-redux';
import Committee from '../Committee/Committee';
import { Route, NavLink } from 'react-router-dom';
import { setPacContributions, setIndividualContributions } from '../../actions';
import { fetchPACContributions, fetchIndividualContributions } from '../../util/apiCalls';
import SearchDisambiguation from '../../Components/SearchDisambiguation.js';


export class App extends Component {

  constructor(props) {
    super(props);
    this.getPacContributions = this.getPacContributions.bind(this);
  }

  componentDidMount = async () => {};

  getPacContributions = async (committee_id) => {
    try {
    let pacContributions = await fetchPACContributions(committee_id);
    this.props.setPacContributions(pacContributions);
    } catch (err) {
      this.setState({ error: err.message })
    }
  }

  getIndividualContributions = async (committee_id) => {
    try {
      let individualContributions = await fetchIndividualContributions(committee_id);
      this.props.setIndividualContributions(individualContributions)
    } catch (err) {
      this.setState({ error: err.message })
    }
  }


  render() {

    return (
      <div className="App">
        <header className="App-header">
        <NavLink to='/'>Home</NavLink>
          <SearchForm />
            <Route path='/candidate/:candidate_id' render={({match}) => {
              return ( <SearchDisambiguation candidate_id={match.params.candidate_id} /> )
          }} />
          
          
        </header>

        <main> 
          <Route path='/candidate/:candidate_id/committee/:id' render={({ match }) => {
            this.props.setPacContributions([]);
            this.getPacContributions(match.params.id);
            this.getIndividualContributions(match.params.id);
            return( <Committee committee_id={match.params.id}/> )
          }} />
        </main>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  candidate: state.candidate,
  committee_id: state.committee_id
})

export const mapDispatchToProps = dispatch => ({
  setPacContributions: pac_contributions => dispatch(setPacContributions(pac_contributions)),
  setIndividualContributions: individual_contributions => dispatch(setIndividualContributions(individual_contributions))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);

// PropTypes
