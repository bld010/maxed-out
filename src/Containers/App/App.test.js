import React from 'react';
import { App } from './App';
import { shallow, mount } from 'enzyme';
import { fetchPACContributions, fetchIndividualContributions } from '../../util/apiCalls';

jest.mock('../../util/apiCalls.js');


let wrapper;
describe('App', () => {


  let mockSetPacContributions = jest.fn();
  let mockSetIndividualContributions = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<App 
    candidate={{}}
    committee_id='mockId'
    setPacContributions={mockSetPacContributions}
    setIndividualContributions={mockSetIndividualContributions}
    />)
  })

  it('should match the snapshot with correct data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('componentDidMount', () => {
    it('should fire getPacContributions with the committee id', async () => {
     
      let mockGetPacContributions = jest.fn().mockReturnValue(Promise.resolve())
      wrapper.instance().getPacContributions = mockGetPacContributions;
      wrapper.update();
      wrapper.instance().componentDidMount();
      await Promise.resolve()
      expect(mockGetPacContributions).toHaveBeenCalledWith('mockId');
    })

    it('should fire getIndividualContributions with the committee id', async () => {
      
      let mockGetPacContributions = jest.fn().mockReturnValue(Promise.resolve())
      wrapper.instance().getPacContributions = mockGetPacContributions;
      let mockGetIndividualContributions = jest.fn().mockReturnValue(Promise.resolve());
      wrapper.instance().getIndividualContributions = mockGetIndividualContributions;
      wrapper.update();
      wrapper.instance().componentDidMount();
      await Promise.resolve()
      expect(mockGetIndividualContributions).toHaveBeenCalledWith('mockId');

    })
  })

  describe('getPacContributions', () => {
    it('should call fetchPACContributions with the committee id', () => {

      fetchPACContributions.mockImplementation(() => {
        return Promise.resolve({ results: [1, 2, 3]})
      })

      wrapper.instance().componentDidMount();
      expect(fetchPACContributions).toHaveBeenCalledWith('mockId');
    })

    it('should call setPacContributions with the results of the fetch', async () => {

      fetchPACContributions.mockImplementation(() => {
        return Promise.resolve({ results: [1, 2, 3]})
      })
      
      wrapper.instance().componentDidMount();
      await Promise.resolve();
      expect(mockSetPacContributions).toHaveBeenCalledWith({results: [1, 2, 3]});
    })

    })
  

  describe('getIndividualContributions', () => {

    it('should call fetchIndividualContributions with the committee id', async () => {

      fetchIndividualContributions.mockImplementation(() => {
        return Promise.resolve({ results: [1, 2, 3]})
      })
      wrapper.instance().getIndividualContributions();
      await Promise.resolve();
      expect(fetchIndividualContributions).toHaveBeenCalledWith('mockId');
    })

    it('should fire setIndividualContributions with the results of the fetch', async () => {

      fetchIndividualContributions.mockImplementation(() => {
        return Promise.resolve({ results: [1, 2, 3]})
      })
      
      wrapper.instance().getIndividualContributions();
      await Promise.resolve();
      expect(mockSetIndividualContributions).toHaveBeenCalled();
    })

    })

  


  describe('mapStateToProps', () => {

  })

  describe('mapDispatchToProps', () => {

  })
})

