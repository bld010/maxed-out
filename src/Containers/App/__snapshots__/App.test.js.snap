// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with correct data passed through 1`] = `
<div
  className="App"
>
  <header>
    <h1>
      Maxed
    </h1>
    <NavLink
      className="nav"
      to="/"
    >
      Home
    </NavLink>
  </header>
  <main>
    <section
      className="search-and-disambiguation"
    >
      <h2>
        Find A Candidate
      </h2>
      <Connect(SearchForm) />
      <Route
        exact={true}
        path="/candidate/:candidate_id"
        render={[Function]}
      />
    </section>
    <section
      className="display"
    >
      <Route
        path="/candidate/:candidate_id/committee/:committee_id"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
    </section>
  </main>
</div>
`;
